protocols/luci-proto-mbim/htdocs/luci-static/resources/protocol/mbim.js

```diff
@@ -0,0 +1,130 @@
+	},
+
+	getOpkgPackage: function() {
+		return 'umbim';
+	},
+
+	isFloating: function() {
+		return true;
+	},
+
+	isVirtual: function() {
+		return true;
+	},
+
+	getDevices: function() {
+		return null;
+	},
+
+	containsDevice: function(ifname) {
+		return (network.getIfnameOf(ifname) == this.getIfname());
+	},
+
+	renderFormOptions: function(s) {
+		var dev = this.getL3Device() || this.getDevice(), o;
+
+		o = s.taboption('general', form.Value, '_modem_device', _('Modem device'));
+		o.ucioption = 'device';
+		o.rmempty = false;
+		o.load = function(section_id) {
+			return callFileList('/dev/').then(L.bind(function(devices) {
+				for (var i = 0; i < devices.length; i++)
+					this.value(devices[i]);
+				return form.Value.prototype.load.apply(this, [section_id]);
+			}, this));
+		};
+
+		o = s.taboption('general', form.Value, 'apn', _('APN'));
+		o.validate = function(section_id, value) {
+			if (!/^[a-zA-Z0-9\-.]*[a-zA-Z0-9]$/.test(value))
+				return _('Invalid APN provided');
+
+			return true;
+		};
+
+		o = s.taboption('general', form.Value, 'pincode', _('PIN'));
+		o.datatype = 'and(uinteger,minlength(4),maxlength(8))';
+
+		o = s.taboption('general', form.ListValue, 'auth', _('Authentication Type'));
+		o.value('both', 'PAP/CHAP');
+		o.value('pap', 'PAP');
+		o.value('chap', 'CHAP');
+		o.value('none', 'NONE');
+		o.default = 'none';
+
+		o = s.taboption('general', form.Value, 'username', _('PAP/CHAP username'));
+		o.depends('auth', 'pap');
+		o.depends('auth', 'chap');
+		o.depends('auth', 'both');
+
+		o = s.taboption('general', form.Value, 'password', _('PAP/CHAP password'));
+		o.depends('auth', 'pap');
+		o.depends('auth', 'chap');
+		o.depends('auth', 'both');
+		o.password = true;
+
+		o = s.taboption('advanced', form.Flag, 'dhcp', _('Use DHCP (not usually needed)'));
+		o.default = o.disabled;
+
+		if (L.hasSystemFeature('ipv6')) {
+			o = s.taboption('advanced', form.Flag, 'ppp_ipv6', _('Enable IPv6 negotiation'));
+			o.ucioption = 'ipv6';
+			o.default = o.disabled;
+		}
+
+		o = s.taboption('advanced', form.Value, 'delay', _('Modem init timeout'), _('Maximum amount of seconds to wait for the modem to become ready'));
+		o.placeholder = '10';
+		o.datatype    = 'min(1)';
+
+		o = s.taboption('general', form.ListValue, 'pdptype', _('PDP Type'));
+		o.value('ipv4v6', 'IPv4/IPv6');
+		o.value('ipv4', 'IPv4');
+		o.value('ipv6', 'IPv6');
+		o.default = 'ipv4v6';
+
+		o = s.taboption('advanced', form.Flag, 'defaultroute',
+			_('Use default gateway'),
+			_('If unchecked, no default route is configured'));
+		o.default = o.enabled;
+
+		o = s.taboption('advanced', form.Value, 'metric',
+			_('Use gateway metric'));
+			o.placeholder = '0';
+		o.datatype = 'uinteger';
+		o.depends('defaultroute', '1');
+
+		o = s.taboption('advanced', form.Flag, 'peerdns',
+			_('Use DNS servers advertised by peer'),
+			_('If unchecked, the advertised DNS server addresses are ignored'));
+		o.default = o.enabled;
```

Comment:
these options are handle by generic, it is likely no need to handle here.

Reply:
maybe we have to handle `mtu` option (Override MTU)

Reply:
and yet it was necessary to add these to the other protocol modules, e.g. fa702c0387dd09a43b1d447e627d118f782ac767

For the most part, this file should be a close copy of the luci-proto-qmi handler.